3604/1/table_output.rb

```diff
@@ -0,0 +1,39 @@
+require 'terminal-table'
+
+# output table with all information about gems
+class TableOutput
+  HASH_FOR_TABLE = {
+    name: '',
+    used_by: 'used by ',
+    watched_by: 'watched by ',
+    stars: 'stars ',
+    forks: 'forks ',
+    contributors: 'contributors ',
+    issues: 'issues '
+  }.freeze
+
+  def table_output(array_gems, count_gems_in_argv)
+    table = Terminal::Table.new
+    table.style = { border_top: false, border_bottom: false }
+    how_many_names_need_to_display(count_gems_in_argv.to_i, array_gems.size).times do |iter|
+      table.add_row transformation_into_a_nice_view(array_gems[iter])
+    end
+    puts table
+  end
+
+  private
+
+  def transformation_into_a_nice_view(hash)
+    hash.map do |object|
+      HASH_FOR_TABLE[object[0]] + object[1]
```

Comment:
object is not so good name. please refer documentation for `.map`. you can pass 2 parameters to block (key and value of each pair in hash). also please give them meaningful names